public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -404,11 +404,31 @@ function getAnnotations(state: DashboardSceneState): AnnotationQueryKind[] {
       },
     };
 
-    // Check if DataQueryKind exists
-    const queryKind = getAnnotationQueryKind(layer.state.query);
-    if (layer.state.query.query?.kind === queryKind) {
+    // For built-in annotations, always use grafana kind but respect target if it exists
+    if (layer.state.query.builtIn) {
       result.spec.query = {
-        kind: queryKind,
+        kind: 'grafana',
+        spec: layer.state.query.target || {
+          type: 'dashboard',
+          limit: 100,
+          matchAny: false,
+          tags: [],
+        },
+      };
+    } else if (layer.state.query.target) {
+      // Transform old target structure to new query structure for non-built-in annotations
+      result.spec.query = {
+        kind: getAnnotationQueryKind(layer.state.query),
+        spec: {
+          ...layer.state.query.target,
+          // Include any datasource-specific query properties (like expr for Prometheus)
+          ...(layer.state.query.expr && { expr: layer.state.query.expr }),
```

Comment:
I think over here we should take _any_ property (not only `expr`) that is not a property of the annotation spec.


Do you know how prometheus ends up adding `expr` to the annotation json, rathen than the target?